lib/at_ex/gateway/base_http.ex

```diff
@@ -49,9 +49,13 @@ defmodule AtEx.Gateway.Base do
       Process results from calling the gateway
       """
       def process_result(result) do
-        with {:ok, res} <- Jason.decode(result) do
-          {:ok, res}
-        else
+        case result do
+          {:ok, %{status: 201} = res} ->
```

Comment:
The issue with this is sometimes the response is `200` and not `201`.

Reply:
This makes its usage in other instances harder, unless we decide to add more process_result methods to handle cases of status is `200`or `201`, or we have the `process_result(result)` take something like `process_result(result, status)` where status can either be `200` or `201`.

Reply:
If that is the case then we can pattern match any 2xx response

```elixir
        case result do
          {:ok, %{status: status} = res} when status > 199 and status < 300-> Jason.decode(res.body)
          {:ok, val} ->
             {:error, %{status: val.status, message: val.body}}
         end
```